src/Microsoft.TemplateSearch.TemplateDiscovery/Nuget/NugetPackInfo.cs

```diff
@@ -7,14 +7,27 @@ namespace Microsoft.TemplateSearch.TemplateDiscovery.Nuget
 {
     public class NugetPackInfo : IDownloadedPackInfo
     {
-        public string VersionedPackageIdentity { get; set; }
+        private IPackInfo _info;
 
-        public string Id { get; set; }
+        internal NugetPackInfo(IPackInfo info, string filePath)
+        {
+            if (string.IsNullOrWhiteSpace(filePath))
+            {
+                throw new System.ArgumentException($"'{nameof(filePath)}' cannot be null or whitespace.", nameof(filePath));
```

Comment:
if the System namespace is referenced,  `System.ArgumentException` could be just `ArgumentException`, unless there is a collision of a type named `ArgumentException` in the referenced namespaces.